src/transformers/models/bark/modeling_bark.py

```diff
@@ -0,0 +1,1402 @@
+        """Turn quantized audio codes into audio array using encodec."""
+
+        fine_output = fine_output.transpose(0, 1)
+        emb = self.codec_model.quantizer.decode(fine_output)
+        out = self.codec_model.decoder(emb)
+        audio_arr = out.squeeze(1)  # squeeze the codebook dimension
+
+        return audio_arr
+
+    @torch.no_grad()
+    def generate_audio(
+        self,
+        input_ids: Optional[torch.Tensor] = None,
+        history_prompt: Optional[Dict[str, torch.Tensor]] = None,
+        **kwargs,
+    ) -> torch.LongTensor:
+        """
+        Generates audio from an input prompt and an additional optional `Bark` speaker prompt.
+
+        Args:
+            input_ids (`Optional[torch.Tensor]` of shape (batch_size, seq_len), *optional*, defaults to None):
+                Input ids. Will be truncated up to 256 tokens. Note that the output audios will be as long as the
+                longest generation among the batch.
+            The last token is `semantic_infer_token`. Note that batch_size is set to 1 to generate one audio per audio. Defaults to None.:
+            history_prompt (`Optional[Dict[str,torch.Tensor]]`, *optional*, defaults to None):
+                Optional `Bark` speaker prompt. Defaults to None. Note that for now, this model takes only one speaker
+                prompt per batch.
+        Returns:
+            torch.LongTensor: Output generated audio.
+
+        Example:
+
+        ```python
+        >>> from transformers import AutoProcessor, BarkModel
+
+
+        >>> processor = AutoProcessor.from_pretrained("ylacombe/bark-small")
+        >>> model = BarkModel.from_pretrained("ylacombe/bark-small")
+
+        >>> # To add a voice preset, you can pass `voice_preset` to `BarkProcessor.__call__(...)`
+        >>> voice_preset = "v2/en_speaker_6"
+
+        >>> inputs = processor("Hello, my dog is cute, I need him in my life", voice_preset=voice_preset)
+
+        >>> audio_array = model.generate_audio(**inputs)
+        >>> audio_array = audio_array.cpu().numpy().squeeze()
+        ```
+        """
+
+        # 1. Generate from the semantic model
+
+        semantic_output = self.generate_text_semantic(
+            input_ids, history_prompt, attention_mask=kwargs.pop("attention_mask", None), **kwargs
+        )
+
+        # 2. Generate from the coarse model
+
+        coarse_output = self.generate_coarse(semantic_output, history_prompt, **kwargs)
+
+        # 3. "generate" from the fine model
+
+        output = self.generate_fine(coarse_output, history_prompt, **kwargs)
+
+        # 4. Decode the output and generate audio array
+
+        audio = self.codec_decode(output)
+
+        return audio
+
+    def can_generate(self) -> bool:
+        """
+        Returns True. Despite not having a `self.generate` method, this model can `generate_audio` and thus needs a
+        BarkGenerationConfig.
+        """
+        return True
+
+
+class SemanticStoppingCriteria(StoppingCriteria):
```

Comment:
Again, like the logits processor this should be in stopping criteria! Shouldn't be in the modelling code. It's the same modelling / generation split that we discussed for the logits processor

Reply:
I remove it for now since it is not used yet